compiler/lib/effects.ml

```diff
@@ -493,12 +494,24 @@ let cps_transform ~live_vars p =
           ; live_vars
           }
         in
-        let k = closure_continuation start in
+        let function_needs_cps =
+          Option.is_some name_opt || not (Addr.Set.is_empty blocks_to_transform)
+        in
+        if not function_needs_cps then wrap_toplevel := false;
```

Comment:
I would have expected the opposite.
I would make it clear that
`not function_needs_cps` can only be true when when looking at the toplevel closure  

```ocaml
let wrap_toplevel = ref false in
...
if toplevel && function_needs_cps then wrap_toplevel := true;
...
```

Reply:
```ocaml
let function_need_cps =
  match name_opt with
  | None -> (* toplevel closure *)
    let need_cps =  not (Addr.Set.is_empty blocks_to_transform) in
    wrap_toplevel := needs_cps
    needs_cps
  | Some _ -> ..
```